pilot/pkg/serviceregistry/kube/controller/testdata/rbac/deny-both-http-tcp.yaml

```diff
@@ -0,0 +1,170 @@
+action: DENY
+groups:
+- rules:
+  - matches:
```

Comment:
Lots of empty seem suspicious?

Reply:
This is since its all L7 attributes, so we want them to become "DENY all".

Maybe I should write my own test cases for this instead of using the AuthzPolicy ones... since they are a bit confusing

Reply:
If we have a deny all with no match, the rest of the policy seems ~irrelevant outside of debug info. 

I think it could also end up making ztunnel go through a bunch of unnecessary matching and then hit one of these last and fail. 

Should we consolidate dupe `DENY ALL`s /avoid sending unnecessary parts?

Reply:
I changed things a bit, should be simpler now. There are still cases where we have empty rules that could be flattened, but i am a bit concerned about introducing bugs that way. I talked with you offline about a simulator, I think it makes sense to implement this once we have that so we can ensure there is no behavioral regressions (rather than relying on human to analyzing a golden file)